com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/CollectionBuilder.java

```diff
@@ -105,24 +95,28 @@ public Task<Void> create(IResource input) throws IOException, CompileExceptionEr
                 .addOutput(input.changeExt(ComponentsCounter.EXT_COL));
         CollectionDesc.Builder builder = CollectionDesc.newBuilder();
         ProtoUtil.merge(input, builder);
+        createSubTasks(builder, taskBuilder);
 
         Map<IResource, Integer> subCollections = new HashMap<>();
         collectSubCollections(builder, subCollections);
         for (IResource subCollection : subCollections.keySet()) {
-            taskBuilder.addInput(subCollection);
             IResource compCounterInput = input.getResource(ComponentsCounter.replaceExt(subCollection)).output();
-            taskBuilder.addInput(compCounterInput);
-            compCounterInputsCount.put(compCounterInput, subCollections.get(subCollection));
+            compCounterInputsCount.put(compCounterInput, 1);
```

Comment:
if there are a few instances of the same go/col, they all will be added as inputs to the collection, so it's anoutgh to specify including each of them only once.